plugins/package-managers/spm/src/main/kotlin/SpmModel.kt

```diff
@@ -33,14 +33,6 @@ import org.ossreviewtoolkit.utils.ort.normalizeVcsUrl
 
 val json = Json { ignoreUnknownKeys = true }
 
-abstract class SpmDependency {
```

Comment:
Commit message nits:
- "aka." without "." at the end.
- Foot notes should have an additional ":", like "[1]:".

Reply:
> The inheritance was used for code re-use

While that's true, I'm not sure whether that was the *only* purpose. I believe it was also used to express that "both libraries and applications are dependencies". This semantic is now lost. It might make sense to keep `SpmDependency` as an `interface` to still express that semantic. I still need to review the following commits to make up my mind about this myself.

Reply:
> It might make sense to keep `SpmDependency` as an `interface` to still express that semantic.

Having things independent does make it easier for me to refactor code. Once done, and it still seems the best thing to do to have a common interface, this can also be done on top in no time.